tools/bazel/compile_commands/compile_commands.py

```diff
@@ -0,0 +1,255 @@
+                        not_found += 1
+                else:
+                    logging.info("Skipping: %s", file)
+                    skipped += 1
+            except ValueError:
+                logging.warning("Wrong compile command: %s", str(arguments))
+    logging.info("%d files added, %d skipped (asm), %d not found",
+                 len(compile_commands), skipped, not_found)
+    return compile_commands
+
+
+def get_full_path(file, directories):
+    """
+    Return full path for a file and try to prepend directories
+    """
+    for directory in directories:
+        full_path = os.path.join(directory, file)
+        if os.path.isfile(full_path):
+            return (full_path, directory)
+    return (False, False)
+
+
+def filter_compile_command(compile_command, output_base):
+    """
+    Remove irrelevant options and change paths
+    """
+    command = compile_command["command"]
+    if output_base:
+        # bazel-out/k8-fastbuild/bin/external/ -> /abs/path/external/
+        command = re.sub(r"( |=)bazel-out\/\S*\/bin\/external\/",
+                         r"\1" + output_base + "/external/",
+                         command)
+    else:
+        # bazel-out/k8-fastbuild/bin/external/ -> ../../external/
+        command = re.sub(r"( |=)bazel-out\/\S*\/bin\/external\/", r"\1../../external/", command)
+    # Remove unsupported options (clang)
+    command = re.sub(r" -MD ", " ", command)
+    command = re.sub(r" -MF \S* ", " ", command)
+    command = re.sub(r" -MT \S* ", " ", command)
+    compile_command["command"] = command
+
+
+def run(build_command, output_file_name):
+    """
+    Run bazel aquaery and generate compile_commands.json:
+    - get bazel info params
+    - run bazel aquery and parse output as JSON data
+    - create compile commands from JSON data:
+      * take only arguments
+      * add: file, directory, command
+      * change to full paths
+      * filter compiler options
+    - save to output file (compile_commands.json)
+    """
+    if not build_command:
+        logging.error("No bazel build command specified")
+        return False
+
+    if not output_file_name:
+        logging.error("No output file is specified")
+        return False
+
+    bazel_info = split_to_list(build_command)[0] + " info "
+    bazel_workspace = run_command(bazel_info + "workspace").rstrip()
+    bazel_output_base = run_command(bazel_info + "output_base").rstrip()
+    bazel_execution_root = run_command(bazel_info + "execution_root").rstrip()
+
+    bazel_work_dir = bazel_execution_root.replace("/execroot/", "/external/")
+    if not os.path.exists(bazel_work_dir):
+        bazel_work_dir = bazel_workspace
+
+    logging.info("Bazel workspace:     %s", bazel_workspace)
+    logging.info("Bazel output base:   %s", bazel_output_base)
+    logging.info("Bazel work dir:      %s", bazel_work_dir)
+
+    logging.info("Bazel build options: %s", build_command)
+    data = bazel_aquery(build_command)
```

Comment:
Use the `cmd` here, so this function will expect a list instead of a string. And there will be no need to call the split function on the argument.

Reply:
Yes, I agree that we can use `shlex.split` however `split_to_list` is more convenient and safer.
I would keep it if you dont mind.